### PR TITLE
qa/suites/upgrade/*-x: skip TestClsRbd.group_snap_{set,remove}

### PR DESCRIPTION
**Pull request by idryomov**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by idryomov** [comment]
cc @nbalacha

**Comment by idryomov** [comment]
jenkins test make check

**Comment by idryomov** [comment]
jenkins test make check

**Review by nbalacha — APPROVED** [review]

**Comment by nbalacha** [comment]
I'd like someone from the rados team to look at the PR before it is merged.

File: qa/suites/upgrade/quincy-x/parallel/workload/rados_api.yaml [hunk 1]
```diff
@@ -9,4 +9,6 @@ workload:
         clients:
           client.0:
             - cls
+        env:
+          CLS_RBD_GTEST_FILTER: '-TestClsRbd.group_snap_set:TestClsRbd.group_snap_remove'
```
Comment (ljflores) [inline qa/suites/upgrade/quincy-x/parallel/workload/rados_api.yaml:13]: How about before each entry add a line pointing to the tracker?
i.e.:
`# See https://tracker.ceph.com/issues/67759`

Should we also add a comment or something to indicate that this test should be added back once the test binaries are compatible again? And perhaps we can open another tracker to track that?

Comment (idryomov) [reply qa/suites/upgrade/quincy-x/parallel/workload/rados_api.yaml:13]: > How about before each entry add a line pointing to the tracker? i.e.: `# See https://tracker.ceph.com/issues/67759`

The tracker and other details can be looked up with `git blame`.

> 
> Should we also add a comment or something to indicate that this test should be added back once the test binaries are compatible again? And perhaps we can open another tracker to track that?

There is no plan to make the test binaries compatible.  It wouldn't be hard to do in this particular case, but in principle these tests aren't meant to be run across versions like that, so I think it's better to just skip the ones that happen to be too strict to pass.

**Review by ljflores — APPROVED** [review]

**Comment by idryomov** [comment]
The job failed on `OSD_DOWN` health check, but it's not related:

https://pulpito.ceph.com/dis-2024-12-04_14:12:29-rados:upgrade-main-distro-default-smithi/

- The OSD was terminated by a signal from cephadm as part of upgrade:
```
2024-12-04T14:44:33.534+0000 7fee95875640 -1 received  signal: Terminated from /run/podman-init -- /usr/bin/ceph-osd -n osd.0 -f --setuser ceph --setgroup ceph --default-log-to-file=false --default-log-to-journald=true --default-log-to-stderr=false  (PID: 1) UID: 0
2024-12-04T14:44:33.534+0000 7fee95875640 -1 osd.0 413 *** Got signal Terminated ***
2024-12-04T14:44:33.534+0000 7fee95875640  0 osd.0 413 Fast Shutdown: - cct->_conf->osd_fast_shutdown = 1, null-fm = 1
2024-12-04T14:44:33.534+0000 7fee95875640 -1 osd.0 413 *** Immediate shutdown (osd_fast_shutdown=true) ***
2024-12-04T14:44:33.534+0000 7fee95875640  0 osd.0 413 prepare_to_stop telling mon we are shutting down and dead
```

- `TestClsRbd` tests ran and succeeded much later:
```
2024-12-04T14:56:02.165 INFO:tasks.workunit.client.0.smithi086.stdout:Note: Google Test filter = -TestClsRbd.group_snap_set:TestClsRbd.group_snap_remove
2024-12-04T14:56:02.165 INFO:tasks.workunit.client.0.smithi086.stdout:[==========] Running 68 tests from 1 test suite.
2024-12-04T14:56:02.165 INFO:tasks.workunit.client.0.smithi086.stdout:[----------] Global test environment set-up.
2024-12-04T14:56:02.165 INFO:tasks.workunit.client.0.smithi086.stdout:[----------] 68 tests from TestClsRbd
...
2024-12-04T14:56:10.065 INFO:tasks.workunit.client.0.smithi086.stdout:[==========] 68 tests from 1 test suite ran. (7900 ms total)
2024-12-04T14:56:10.065 INFO:tasks.workunit.client.0.smithi086.stdout:[  PASSED  ] 68 tests.
```

**Comment by idryomov** [comment]
Filed https://tracker.ceph.com/issues/69135 to track the above -- likely a missing ignorelist.

**Pull request merged**